python_modules/dagster-graphql/dagster_graphql/implementation/fetch_asset_checks.py

```diff
@@ -46,12 +48,23 @@ def fetch_asset_checks(
 ) -> Union[
     GrapheneAssetCheckNeedsMigrationError,
     GrapheneAssetCheckNeedsUserCodeUpgrade,
+    GrapheneAssetCheckNeedsAgentUpgradeError,
     GrapheneAssetChecks,
 ]:
-    if not graphene_info.context.instance.event_log_storage.supports_asset_checks:
+    asset_check_support = graphene_info.context.instance.get_asset_check_support()
+    if asset_check_support == AssetCheckInstanceSupport.NEEDS_MIGRATION:
         return GrapheneAssetCheckNeedsMigrationError(
             message="Asset checks require an instance migration. Run `dagster instance migrate`."
         )
+    elif asset_check_support == AssetCheckInstanceSupport.NEEDS_AGENT_UPGRADE:
+        return GrapheneAssetCheckNeedsAgentUpgradeError(
+            "Asset checks require an agent upgrade to 1.5.0 or greater."
```

Comment:
A little weird that we need to special case this option that's not in the OSS codebase. Thoughts on naming the instance method something like `get_supports_asset_check_error` and having the cloud code instantiate this object?

Reply:
What would the return type be? The exact string we show in the frontend? Currently it has a bit of formatting we'd have to replicate

Reply:
I was imagining it would be the error object but string could work too.

Reply:
agreed it's weird to reference agents in oss, but I think it's worth it to get nice formatting like we have for the current migration error https://github.com/dagster-io/dagster/blob/johann/10-05-gate_asset_checks_on_agent_version/js_modules/dagster-ui/packages/ui-core/src/assets/asset-checks/AssetCheckDetailModal.tsx#L302-L319

To do this properly I think we'd extend the gql layer and js in cloud, but that's non trivial